code/datums/emote_system/emote.dm

```diff
@@ -0,0 +1,223 @@
+	var/count
+	for(var/i = 1, i<=lentext(pathString), i++)
+		var/char = copytext(pathString, i, i+1)
+		if(char == "/")
+			count++
+			if(count == 4)
+				baseLevel = 0
+				break
+
+/datum/emote/proc/doEmote(var/mob/user, var/param)
+	if(cooldown)
+		if(handle_emote_CD(user))
+			return 0
+
+	var/message = createMessage(user, param)
+
+	if(canTarget && param)
+		param = getTarget(user, param)
+
+	if(message)
+		message = addTarget(user, param, message)
+		message += "</span>"
+		outputMessage(user, message)
+
+	if(audible)
+		if(!(user.mind && user.mind.miming))
+			playSound(user, vol)
+
+	doAction(user, param)
+	return 1
+
+// for things that the emote does that aren't text or sound based
+/datum/emote/proc/doAction(var/mob/user, var/param)
+	return
+
+// returns the reason the user can't currently do the emote
+/datum/emote/proc/prevented(var/mob/user)
+	if(user.stat == DEAD)
+		return "you are dead"
+	if(restrained && user.restrained())
+		return "you are restrained"
+
+// return 1 if this emote can be used by this type of user
+/datum/emote/proc/available(var/mob/user)
+	return
+
+/datum/emote/proc/createMessage(var/mob/user, var/param)
+	if(!text)
+		return
+	var/message
+	if(audible && user.mind && user.mind.miming)
+		message = mimeMessage(user)
+	else if(audible && user.is_muzzled())
+		message = muzzleMessage(user)
+	else if(takesParam && param)
+		message = paramMessage(user, param)
+	else
+		message = "<span class = '[spanClass]'>[startText][user] [text]"
+	return message
+
+/datum/emote/proc/mimeMessage(var/mob/user)
+	if(!mimeText)
+		return
+
+	var/message = "<span class = '[spanClass]'>[startText][user] [mimeText]"
+	return message
+
+/datum/emote/proc/muzzleMessage(var/mob/user)
+	var/message = "<span class = '[spanClass]'>makes a [muzzledNoise]noise"
+	return message
+
+// if the emote takes a non target parameter, set up  and return the with parameter version in here
+/datum/emote/proc/paramMessage(var/mob/user, var/param)
+	return
+
+/datum/emote/proc/addTarget(var/mob/user, var/target, var/message)
+	if(!canTarget)
+		return message
+	target = getTarget(user, target)
+	if(!target)
+		return message
+	if(ismob(target))
+		message += " [targetText] [target]"
+	else
+		message += " [targetText] \the [target]"
+	return message
+
+/datum/emote/proc/getTarget(var/mob/user, var/target)
+	if(!target)
+		return
+	if(targetMob)
+		return getMobTarget(user, target)
+	else
+		return getAtomTarget(user, target)
+
+/datum/emote/proc/getMobTarget(var/mob/user, var/target)
+	for (var/mob/M in view(null, user))
+		if (target == M.name)
+			return M
+
+/datum/emote/proc/getAtomTarget(var/mob/user, var/target)
+	if (target)
+		for (var/atom/A as mob|obj|turf in view(null, user))
+			if (target == A.name)
+				return A
+
+/datum/emote/proc/outputMessage(var/mob/user, var/message)
+	if(!message)
+		return
+	log_emote("[user.name]/[user.key] : [message]")
+	if(audible)
+		user.audible_message(message)
+	else
+		user.visible_message(message)
```

Comment:
@tigercat2000 
Just looked into this (it was late last night, so took it at face value). Not sure what you mean by will only show it to the user. The first argument for visible_message is the one that gets shown to everyone (including you if there's no self message), so everyone will see it.
Currently emotes don't use self, blind or deaf message args, however, I would agree that they should... I'll add it to the to do list